src/charm.py

```diff
@@ -35,13 +52,24 @@ def __init__(self, *args):
         }
         self._context = {"namespace": self._namespace, "app_name": self._name}
 
+        self.dashboard_provider = GrafanaDashboardProvider(self)
+
         self.framework.observe(self.on.install, self._on_install)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(
             self.on.training_operator_pebble_ready,
             self._on_training_operator_pebble_ready,
         )
 
+        monitoring_events = [
+            self.on["monitoring"].relation_changed,
```

Comment:
Shall we agree on a naming convention? I see `metrics-endpoint` and `monitoring`. Any suggestions here @simskij ?

Reply:
Changed to `metrics-endpoint` as discussed on different PRs